src/test/java/com/github/bibsysdev/nva/cristin/person/CristinPersonsHandlerTest.java

```diff
@@ -0,0 +1,227 @@
+    private final Environment environment = new Environment();
+    private CristinApiClient cristinApiClientStub;
+    private Context context;
+    private ByteArrayOutputStream output;
+    private CristinPersonsHandler handler;
+
+    @Test
+    public void handlerReturnsOkWhenInputContainsName() throws Exception {
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+    }
+
+    @Test
+    public void handlerIgnoresErrorsWhenTryingToEnrichPersonInformation() throws Exception {
+        cristinApiClientStub = Mockito.spy(cristinApiClientStub);
+        doThrow(new IOException()).when(cristinApiClientStub).getPerson(any(), any());
+        handler = new CristinPersonsHandler(cristinApiClientStub, environment);
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+        assertEquals(MediaType.APPLICATION_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    public void handlerThrowsInternalErrorWhenQueryingPersonsFails() throws Exception {
+        cristinApiClientStub = Mockito.spy(cristinApiClientStub);
+        doThrow(new IOException()).when(cristinApiClientStub).queryAndEnrichPersons(any(), any());
+        handler = new CristinPersonsHandler(cristinApiClientStub, environment);
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+
+        assertEquals(HttpURLConnection.HTTP_INTERNAL_ERROR, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    public void handlerThrowsBadRequestWhenMissingNameQueryParameter() throws IOException {
+        InputStream input = requestWithQueryParameters(Map.of(LANGUAGE_QUERY_PARAMETER, LANGUAGE_NB));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<PersonsWrapper> response = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, response.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    public void handlerSetsDefaultValueForMissingOptionalLanguageParameterAndReturnOk() throws Exception {
+        InputStream input = requestWithQueryParameters(Map.of(NAME_QUERY_PARAMETER, NAME_MATTIAS));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<PersonsWrapper> response = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+        assertEquals(MediaType.APPLICATION_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    public void handlerReceivesAllowOriginHeaderValueFromEnvironmentAndPutsItOnResponse() throws Exception {
```

Comment:
follow the naming convention: https://unit.atlassian.net/wiki/spaces/NVAP/pages/1587871753/Code+style+and+conventions+in+NVA+Backend